src/child/actions/sidebar.js

```diff
@@ -16,3 +17,27 @@ export const selectFavourites = createActionCreator(() => ({
         action: 'Favourites'
     }
 }));
+
+const windowReopened = createActionCreator((reopenWindowName) => ({
+    type: ACTION_TYPES.REOPEN_WINDOW,
+    reopenWindowName
+}));
+
+const openWindowError = createActionCreator((reopenWindowName) => ({
+    type: ACTION_TYPES.OPEN_WINDOW_ERROR,
+    reopenWindowName
+}));
+
+export function reopenWindow(windowName) {
+    return dispatch => {
+        const childWindow = new fin.desktop.Window(
```

Comment:
Might it be better to follow the pattern used when dragging out a window, to send a message to the parent service to open the window?

As it is, does this mean a reopened window would be child of a child (or is the window creation run in the parent)? It would be more consistent and I think it would avoid having to store a global reference to the store on the child windows.